src/agent.ts

```diff
@@ -247,10 +263,26 @@ export class Agent {
    */
   public async getId(): Promise<string> {
     if (!this.id) {
-      const query = `SELECT Id FROM BotDefinition WHERE DeveloperName = '${this.options.name}'`;
-      this.id = (await this.options.connection.singleRecordQuery<{ Id: string }>(query)).Id;
+      await this.getBotMetadata();
+    }
+    return this.id!; // getBotMetadata() ensures this.id is not undefined
+  }
+
+  /**
+   * Queries BotDefinition for the bot metadata and assigns:
+   * 1. this.id
+   * 2. this.name
+   * 3. this.botMetadata
+   */
+  public async getBotMetadata(): Promise<BotMetadata> {
+    if (!this.botMetadata) {
+      const whereClause = this.id ? `Id = '${this.id}'` : `DeveloperName = '${this.name as string}'`;
```

Comment:
TS doesn't complain without the `as string` assertion, maybe use `ensureString` if you want extra safety?

Reply:
It was complaining for me.  Happy to remove it if that's no longer the case.

Reply:
Oh, it's not a TS complaint.  It's the linter.